### ISSUE TITLE
Torchtext not compatible with torch version 1.9.0.dev20210601+cu113

### ISSUE THREAD

**Issue by JVedant** [pytorch/text#1320]
## ❓ Questions and Help

**Description**
As the cuda version I'm using is 11.2, I've installed torch version "1.9.0.dev20210601+cu113" (nightly build) and when I'm installing torchtext, it's changing the version of torch to 1.8.1 and as it's incompatible with cuda 11.2 the torch module doesn't work (It doesn't find an appropriate version of CUDA kernel to run on).

Is there any way of installing torchtext nightly build which works with torch "1.9.0.dev20210601+cu113" ?

**Comment by parmeet** [comment]
cc: @seemethere @malfet

**Comment by malfet** [comment]
torchtext is CPU only library and as such does not have dependencies on a particular CUDA version.
@parmeet it looks like although binary builds are succeeded they are not getting published to https://anaconda.org/pytorch-nightly/torchtext/files
@JVedant are you trying to install it using pip or conda? Can you share exact command you are running
P.S. Please note, that neither CUDA-11.3 nor 11.2 are officially supported by PyTorch at the moment. To get list of supported CUDA configurations please refer to https://pytorch.org/get-started/locally/

**Comment by JVedant** [comment]
yeah i agree that torchtext is CPU based but it's used with torch to train neural nets where the inputs are based on torchtext so as the torch version changes there are no ways to train a model. Hence, only 1 of the two can be used at a time.

Also i'm using pip to install torchtext. The nightly version of torch can be installed using 
`pip3 install --pre torch torchvision torchaudio -f https://download.pytorch.org/whl/nightly/cu113/torch_nightly.html`

followed by `pip3 install torchtext` to install torchtext.

when installing torchtext using the above mentioned command it will uninstall torch version "1.9.x nightly build" and install torch version 1.8.x which is again not compatible with cuda 11.2.

maybe this can be resolved if the dependencies of torchtext can be changed to just any torch version and not any specific version (here, torch 1.8.x).

PS: please consider the install sequence as initially the first command is executed to get the nightly build of torch and then torchtext is installed.

**Comment by parmeet** [comment]
Note that `pip3 install torchtext` will give you the release version (not the nightly). Since the latest torchtext released version is 0.9.1, it will change the torch version back to 1.8.1

You may try installing torchtext  nightly together with torch

`pip3 install --pre --upgrade torch torchvision torchaudio torchtext -f https://download.pytorch.org/whl/nightly/cu113/torch_nightly.html`

**Comment by JVedant** [comment]
I tried to install using the above mentioned command (using pip), but it seems that now the command installs the stable version of torch i.e. 1.8.0 but the torchtext version it installs is 0.10.0

**Comment by JVedant** [comment]
Hi @parmeet , I've tried the above mentioned method multiple times but every time Torchtext is installed, it just changes the version of pytorch to 1.8.x. 

Is there any other method to work with it ?

**Comment by malfet** [comment]
`pip3 install -pre torchtext -f https://download.pytorch.org/whl/nightly/cpu/torch_nightly.html` (without `--pre` flag pip will roll one back to latest released binary) And you can add `--no-deps` if it still attempts to downgrade PyTorch for you.

**Comment by Deepakn8861** [comment]
> ## ❓ Questions and Help
> **Description** As the cuda version I'm using is 11.2, I've installed torch version "1.9.0.dev20210601+cu113" (nightly build) and when I'm installing torchtext, it's changing the version of torch to 1.8.1 and as it's incompatible with cuda 11.2 the torch module doesn't work (It doesn't find an appropriate version of CUDA kernel to run on).
> 
> Is there any way of installing torchtext nightly build which works with torch "1.9.0.dev20210601+cu113" ?

if you try to install pytorch cuda it is will not work i tried this and worked
# CPU Only
`conda install pytorch==2.3.1 torchvision==0.18.1 torchaudio==2.3.1 cpuonly -c pytorch` and installed `pip install torchtext` it worked.
